styx-scheduler-service/src/main/java/com/spotify/styx/api/SchedulerResource.java

```diff
@@ -252,21 +253,31 @@ private StatusType eventInjectorHelper(Event event) {
       return Response.forStatus(BAD_REQUEST.withReasonPhrase(e.getMessage()));
     }
 
-    // Verifying active
-    if (stateManager.isActiveWorkflowInstance(workflowInstance)) {
-      return Response.forStatus(
-          BAD_REQUEST.withReasonPhrase("The specified instance is already "
-                                       + "active in the scheduler"));
-    }
-
     // Verifying future
     if (instant.isAfter(time.get())) {
       return Response.forStatus(BAD_REQUEST.withReasonPhrase(
           "Cannot trigger an instance of the future"));
     }
 
     final String triggerId = randomGenerator.generateUniqueId(AD_HOC_CLI_TRIGGER_PREFIX);
-    triggerListener.event(workflow, Trigger.adhoc(triggerId), instant);
+    final CompletionStage<Void> triggered = triggerListener.event(workflow, Trigger.adhoc(triggerId), instant);
+
+    // TODO: return future instead of blocking
+    try {
+      triggered.toCompletableFuture().get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    } catch (ExecutionException e) {
+      final Throwable cause = e.getCause();
```

Comment:
In the test, `shouldFailTriggerWFIfAlreadyActive`, we use `Throwables.getRootCause` - should that be the same here?

Reply:
Turns out `shouldFailTriggerWFIfAlreadyActive` test was not exactly right. Fixed.